src/main/java/uk/gov/companieshouse/filinghistory/api/service/ValidatorFactory.java

```diff
@@ -18,23 +17,26 @@ public class ValidatorFactory {
     private final TopLevelPutRequestValidator topLevelPutRequestValidator;
     private final AnnotationPutRequestValidator annotationPutRequestValidator;
     private final AssociatedFilingPutRequestValidator associatedFilingPutRequestValidator;
+    private final ResolutionPutRequestValidator resolutionPutRequestValidator;
 
-    public ValidatorFactory(TopLevelPutRequestValidator topLevelPutRequestValidator, AnnotationPutRequestValidator annotationPutRequestValidator, AssociatedFilingPutRequestValidator associatedFilingPutRequestValidator) {
+    public ValidatorFactory(TopLevelPutRequestValidator topLevelPutRequestValidator,
+            AnnotationPutRequestValidator annotationPutRequestValidator,
+            AssociatedFilingPutRequestValidator associatedFilingPutRequestValidator,
+            ResolutionPutRequestValidator resolutionPutRequestValidator) {
         this.topLevelPutRequestValidator = topLevelPutRequestValidator;
         this.annotationPutRequestValidator = annotationPutRequestValidator;
         this.associatedFilingPutRequestValidator = associatedFilingPutRequestValidator;
+        this.resolutionPutRequestValidator = resolutionPutRequestValidator;
     }
 
     public Validator<InternalFilingHistoryApi> getPutRequestValidator(TransactionKindEnum kind) {
-        LOGGER.debug("Getting validator for [%s] transaction kind".formatted(kind.getValue()), DataMapHolder.getLogMap());
+        LOGGER.debug("Getting validator for [%s] transaction kind".formatted(kind.getValue()),
+                DataMapHolder.getLogMap());
         return switch (kind) {
             case TOP_LEVEL -> topLevelPutRequestValidator;
             case ANNOTATION -> annotationPutRequestValidator;
             case ASSOCIATED_FILING -> associatedFilingPutRequestValidator;
-            case RESOLUTION -> {
-                LOGGER.error("Invalid transaction kind: %s".formatted(kind.getValue()), DataMapHolder.getLogMap());
-                throw new InvalidTransactionKindException("Invalid transaction kind: %s".formatted(kind.getValue()));
-            }
+            case RESOLUTION -> resolutionPutRequestValidator;
```

Comment:
Same as comment for AbstractTransactionMapperFactory above.

Reply:
Same reason as to above as why I didn't implement it originally, intellij seems to complain for me. Implemented now.

Reply:
resolved in call.